internal/operators/operators.go

```diff
@@ -0,0 +1,258 @@
+
+	// PerGadgetParams will return params (required) per gadget instance of the operator
+	PerGadgetParams() params.Params
+
+	// Dependencies can list other operators that this operator depends on
+	Dependencies() []string
+
+	// CanOperateOn should test whether the operator supports the given gadget. Init has not
+	// necessarily been called at this point.
+	CanOperateOn(gadgets.Gadget) bool
+
+	// Init allows the operator to initialize itself
+	Init(params params.Params) error
+
+	// Cleanup allows the operator to clean up stuff prior to exiting
+	Cleanup() error
+
+	// PreGadgetRun is called before a gadget is started; the operator must return something that implements operator
+	// This is useful to create a context for an operator by wrapping it.
+	// Params given here are the ones returned by PerGadgetParams()
+	PreGadgetRun(Runner, any, params.Params) (Operator, error)
+
+	// PostGadgetRun is called on the operator that was returned from PrepareTrace after a
+	// gadget was stopped
+	PostGadgetRun() error
+
+	// EnrichEvent is called on the operator returned by StartTrace and should perform
+	// the actual enrichment
+	EnrichEvent(any) error
+}
+
+type Operators []Operator
+
+// KubernetesFromMountNSID is a typical kubernetes operator interface that adds node, pod, namespace and container
+// information given the MountNSID
+type KubernetesFromMountNSID interface {
+	ContainerInfoSetters
+	GetMountNSID() uint64
+}
+
+type ContainerInfoSetters interface {
+	SetContainerInfo(pod, namespace, container string)
+	SetNode(string)
+}
+
+var operators = map[string]Operator{}
+
+type operatorWrapper struct {
+	Operator
+	initOnce    sync.Once
+	initialized bool
+}
+
+func (e *operatorWrapper) Init(params params.Params) (err error) {
+	e.initOnce.Do(func() {
+		err = e.Operator.Init(params)
+		e.initialized = true
+	})
+	return err
+}
+
+// RegisterOperator adds a new operator to the registry
+func RegisterOperator(operator Operator) {
+	if _, ok := operators[operator.Name()]; ok {
+		panic(fmt.Errorf("operator already registered: %q", operator.Name()))
+	}
+	operators[operator.Name()] = operator
+}
+
+// OperatorsParamsCollection returns a collection of params of all registered operators
+func OperatorsParamsCollection() params.ParamsCollection {
+	pc := make(params.ParamsCollection)
+	for _, operator := range operators {
+		pc[operator.Name()] = operator.Params()
+	}
+	return pc
+}
+
+// GetOperatorsForGadget checks which operators can work with the given gadgets and returns a collection
+// of them
+func GetOperatorsForGadget(gadget gadgets.Gadget) Operators {
+	out := make(Operators, 0)
+	for _, e := range operators {
+		if e.CanOperateOn(gadget) {
+			out = append(out, e)
+		}
+	}
+	out, err := SortOperators(out)
+	if err != nil {
+		panic(fmt.Sprintf("sorting operators: %v", err))
+	}
+	return out
+}
+
+// InitAll initialized all registered operators using their respective params
+func (e Operators) InitAll(pc params.ParamsCollection) error {
```

Comment:
All functions are iterating on the operators, then just `Init()` should be fine.